file.go

```diff
@@ -293,15 +293,25 @@ func (f *File) reload(s dataSource) error {
 
 // Reload reloads and parses all data sources.
 func (f *File) Reload() (err error) {
-	for _, s := range f.dataSources {
+	for i, s := range f.dataSources {
 		if err = f.reload(s); err != nil {
+			// Notice: if ShortCircuit and Loose are both true, ShortCircuit has priority over Loose.
+			// Return err when all configuration sources fail to load.
```

Comment:
```suggestion
```

These lines seems no longer relevant:
1. It is already obvious ShortCircuit takes priority here.
2. No longer returns an error for it.